tst/regression/utils/test_case.py

```diff
@@ -106,20 +106,10 @@ def __init__(self,run_test_path,**kwargs):
         self.parameters.output_path = output_path
         self.parameters.test_path = test_path
         self.parameters.mpi_cmd = mpi_executable
+        self.parameters.mpi_ranks = kwargs.pop('mpirun_ranks')
         self.parameters.mpi_opts = kwargs.pop('mpirun_opts')
-       
-        argstrings = ['-np','-n']
-        if len(set(argstrings) & set(self.parameters.mpi_opts)) > 1:
-          print('Warning! You have set both "-n" and "-np" in your MPI options.')
-          print(self.parameters.mpi_opts)
-        for s in argstrings:
-          if s in self.parameters.mpi_opts:
-            index = self.parameters.mpi_opts.index(s)
-            if index < len(self.parameters.mpi_opts) - 1:
-              try:
-                self.parameters.num_ranks = int(self.parameters.mpi_opts[index+1])
-              except ValueError:
-                pass
+        if len(self.parameters.mpi_ranks) == 2:
+            self.parameters.num_ranks = int(self.parameters.mpi_ranks[1])
```

Comment:
When will num_ranks be different from mpi_ranks? If the array is empty I see it but what about if mpi_ranks == 4?

Reply:
As mentioned in the comment above mpirun_ranks will always contain two entries. One for the flag and one for the number.

Reply:
Oh, I completely parsed over len, disregard my comment.